tmt/steps/execute/internal.py

```diff
@@ -242,13 +242,18 @@ def _test_environment(
         assert self.parent is not None
         assert isinstance(self.parent, tmt.steps.execute.Execute)
 
+        serial_number = str(invocation.test.serial_number)
+        assert self.parent.plan.my_run is not None
+        iteration_id = f"{self.parent.plan.my_run.unique_id}-{serial_number}"
```

Comment:
One nitpick-ish comment: the only purpose of these two names is to be used exactly once... Every other envvar on the list is constructed from primary sources, and it's not the worst list we can find in tmt sources :)

```python
        environment['TMT_TEST_SERIAL_NUMBER'] = EnvVarValue(str(invocation.test.serial_number))
        environment['TMT_TEST_ITERATION_ID'] = EnvVarValue(f"{self.parent.plan.my_run.unique_id}-{invocation.test.serial_number}")
```

Reply:
Fixed in 23de3ba62efb2325a970fec35960f7f3d4891b78.

Reply:
Thank you.